LDAP-Auth/LDAPAuthenticationProviderPlugin.cs

```diff
@@ -210,11 +213,65 @@ public Task ChangePassword(User user, string newPassword)
 
         private static bool LdapClient_UserDefinedServerCertValidationDelegate(
             object sender,
-            System.Security.Cryptography.X509Certificates.X509Certificate certificate,
-            System.Security.Cryptography.X509Certificates.X509Chain chain,
+            X509Certificate certificate,
+            X509Chain chain,
+            SslPolicyErrors sslPolicyErrors)
+        {
+            if ((sslPolicyErrors & SslPolicyErrors.RemoteCertificateNameMismatch) > 0 ||
+                (sslPolicyErrors & SslPolicyErrors.RemoteCertificateNotAvailable) > 0)
+            {
+                _logger.LogWarning("Server certificate is not valid for remote domain");
+                return false;
+            }
+
+            X509Chain rootChain = new X509Chain
```

Comment:
```suggestion
            using var rootChain = new X509Chain
```

`X509Chain` implements `IDisposable`. Please test after changing to make sure it still works